rapids_build_backend/config.py

```diff
@@ -12,10 +12,10 @@ class Config:
     # required) and whether it may be overridden by an environment variable or a config
     # setting.
     config_options = {
+        "allow-nightly-deps": (True, True),
```

Comment:
So do you think we should default this to true but still specify in every project's pyproject.toml that it's true? Or should it default to false? The former is probably the most "correct" while also conveying the most information to readers of pyproject.toml, it's just also redundant.

Reply:
Default of `None` (required for validation) would be another solution, specifically because we want to differ from “normal expectations” by default. However I am alright with defaulting to True for now, and specifying it in each package for explicitness. We’ll be able to change the default later if needed.